bin/behat-prepare.sh

```diff
@@ -22,125 +23,191 @@ fi
+cp -r "$PREPARE_DIR"/private/simplesamlphp/config "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/templates "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/cert "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/metadata "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/routing "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/attributemap "$PREPARE_DIR"/simplesaml/
+cp -r "$PREPARE_DIR"/private/simplesamlphp/lib "$PREPARE_DIR"/simplesaml/
+
+# Modify the include...
+sed -i "s|dirname(__FILE__, 2) . '/src/_autoload.php'|__DIR__ . '/src/_autoload.php'|" "$PREPARE_DIR/simplesaml/_include.php"
 
 ###
 # Push files to the environment
 ###
-cd $PREPARE_DIR
 git add private wp-content simplesaml
 git config user.email "wp-saml-auth@getpantheon.com"
 git config user.name "Pantheon"
 git commit -m "Include SimpleSAMLphp and its configuration files"
 git push
 
 # Sometimes Pantheon takes a little time to refresh the filesystem
-terminus build:workflow:wait $TERMINUS_SITE.$TERMINUS_ENV
+terminus build:workflow:wait "$TERMINUS_SITE"."$TERMINUS_ENV"
 
 ###
 # Set up WordPress, theme, and plugins for the test run
 ###
 # Silence output so as not to show the password.
 {
-  terminus wp $SITE_ENV -- core install --title=$TERMINUS_ENV-$TERMINUS_SITE --url=$PANTHEON_SITE_URL --admin_user=$WORDPRESS_ADMIN_USERNAME --admin_email=$WORDPRESS_ADMIN_EMAIL --admin_password=$WORDPRESS_ADMIN_PASSWORD
+  terminus wp "$SITE_ENV" -- core install --title="$TERMINUS_ENV"-"$TERMINUS_SITE" --url="$PANTHEON_SITE_URL" --admin_user="$WORDPRESS_ADMIN_USERNAME" --admin_email="$WORDPRESS_ADMIN_EMAIL" --admin_password="$WORDPRESS_ADMIN_PASSWORD"
 } &> /dev/null
-terminus wp $SITE_ENV -- option update home "https://$PANTHEON_SITE_URL"
-terminus wp $SITE_ENV -- option update siteurl "https://$PANTHEON_SITE_URL"
-terminus wp $SITE_ENV -- plugin activate wp-native-php-sessions wp-saml-auth
-terminus wp $SITE_ENV -- theme activate $TERMINUS_SITE
-terminus wp $SITE_ENV -- rewrite structure '/%year%/%monthnum%/%day%/%postname%/'
+terminus wp "$SITE_ENV" -- option update home "https://$PANTHEON_SITE_URL"
+terminus wp "$SITE_ENV" -- option update siteurl "https://$PANTHEON_SITE_URL"
+terminus wp "$SITE_ENV" -- plugin activate wp-native-php-sessions wp-saml-auth
+terminus wp "$SITE_ENV" -- theme activate "$TERMINUS_SITE"
+terminus wp "$SITE_ENV" -- rewrite structure '/%year%/%monthnum%/%day%/%postname%/'
+# Create writeable directories in /files (aka wp-content/uploads) that SimpleSAMLphp might need.
+terminus wp "$SITE_ENV" -- eval '
+    $dirs = [
+        WP_CONTENT_DIR . "/uploads/simplesaml/log",
+        WP_CONTENT_DIR . "/uploads/simplesaml/data",
+        WP_CONTENT_DIR . "/uploads/simplesaml/tmp",
+    ];
+    foreach ($dirs as $dir) {
+        if ( ! file_exists($dir) ) {
+            mkdir($dir, 0775, true);
+        }
+    }
+'
```

Comment:
lol this seems hacky.

Reply:
it's all hacky. but if these folders don't exist simplesaml errors. this way we also know that they're writeable by WordPress